memtx: fix a crash when MVCC and func index are used (#11094)
tarantool/tarantool: changelogs/unreleased/memtx-mvcc-forbid-multikey.md

## bugfix/memtx

>  * Tarantool allowed to create multikey indexes with memtx MVCC enabled,
>    but they were not supported - it led to a crash or a panic. Now
>    Tarantool raises an error when one tries to create such index with
>    memtx MVCC enabled (gh-6385).

Reviewer: ```suggestion
 * Tarantool allowed to create multikey indexes with memtx MVCC enabled,
   but they were not supported. This led to a crash or a panic. Now
   Tarantool raises an error when one tries to create an index of the kind
   with memtx MVCC enabled (gh-6385).
```
